### PR TITLE
CuML for fast clustering and dim.red.

### PR DESCRIPTION
**Pull request by akalino**

### CODE REVIEW

File: src/eval/BaselineEvals.py [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+import pickle
+import seaborn as sns
+import numpy as np
+import time
+import torch
+import os
+import sys
+import cuml
+
+from cuml.manifold import TSNE, UMAP
```
Comment (akalino) [inline src/eval/BaselineEvals.py:13]: check if cuml is installed, else use sklearn
from sklearn.manifold import TSNE
import UMAP

File: src/eval/BaselineEvals.py [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+    sns.scatterplot(x="component_1", y="component_2",
+                    data=_proj, hue=_rel_labels)
+    plt.title("TSNE Triple2Vec of {}".format(_ds_name))
+    plt.show()
+
+
+def apply_proj(_wvs, _type):
+    RS = 17
+    _wvs = _wvs.numpy()
+    cuml.set_global_output_type('numpy')
```
Comment (akalino) [inline src/eval/BaselineEvals.py:194]: comment this out if not using cuml

File: src/eval/run_eval.py [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+            nn_dict = pickle.load(f)
+            print('Loaded local neighbor index')
+    except FileNotFoundError:
+        print('No local index found, running Pinecone queries')
+        nn_dict = bng(fil_vecs, index, 20, 0.7)
+        with open('neigh_{}.pkl'.format(sent_mod_name), 'wb') as f:
+            pickle.dump(dict(nn_dict), f)
+
+    g = nx.DiGraph(nn_dict).to_undirected()
+    print('Graph loaded, running community detection...')
```
Comment (akalino) [inline src/eval/run_eval.py:228]: print(nx.is_connected(g))
